src/db/migrate.rs

```diff
@@ -1,11 +1,29 @@
 //! Database migrations
 
-use db::connect_db;
 use error::Result as CratesfyiResult;
-use postgres::error::Error as PostgresError;
-use postgres::transaction::Transaction;
+use postgres::{Connection, transaction::Transaction, Error as PostgresError};
 use schemamama::{Migration, Migrator, Version};
 use schemamama_postgres::{PostgresAdapter, PostgresMigration};
+use std::rc::Rc;
+
+
+enum ApplyMode {
+    Permanent,
+    Temporary,
+}
+
+struct MigrationContext {
+    apply_mode: ApplyMode,
```

Comment:
I saw lots of calls to `.clone()`, I pushed a commit that makes this `Copy` instead. A boolean is about as cheap to copy as it gets.

Reply:
Yeah, at the moment your approach is way better. I chose `Rc<MigrationContext>` to future-proof if we add non-`Copy` context items, but we can think about that when we need those.